src/cdk-experimental/listbox/listbox.ts

```diff
@@ -6,26 +6,112 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+import {
+  ContentChildren,
+  Directive,
+  ElementRef,
+  Input,
+  QueryList
+} from '@angular/core';
 
+/**
+ * Directive that applies interaction patterns to an element following the aria role of option.
+ * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
+ * value is built in.
+ */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
+    '[attr.aria-selected]': '_selected',
+    '[attr.data-optionid]': '_optionId',
   }
 })
 export class CdkOption {
+  private _selected: boolean | null = null;
+  private _optionId: string;
+
+  /** Whether the option is selected or not */
+  @Input()
+  get selected(): boolean | null {
+    return this._selected;
+  }
+  set selected(value: boolean | null) {
+    this._selected = value;
+  }
+
+  constructor(private readonly _el: ElementRef) {
+    this.setOptionId(`cdk-option-${_uniqueIdCounter++}`);
+  }
 
+  /** Sets the optionId to the given id */
+  setOptionId(id: string): void {
+    this._optionId = id;
+  }
+
+  /** Returns the optionId of this option */
+  getOptionId(): string {
+    return this._optionId;
+  }
+
+  /** Returns an ElementRef of this option */
+  getElementRef(): ElementRef {
+    return this._el;
+  }
 }
 
+let _uniqueIdCounter = 0;
+
+/**
+ * Directive that applies interaction patterns to an element following the aria role of listbox.
+ * Typically CdkOption elements are placed inside the listbox. Logic to handle keyboard navigation,
+ * selection of options, active options, and disabled states is built in.
+ */
 @Directive({
   selector: '[cdkListbox]',
   exportAs: 'cdkListbox',
   host: {
     role: 'listbox',
+    '(click)': 'onClickUpdateSelectedOption($event)'
   }
 })
 export class CdkListbox {
 
+  /** A query list containing all CdkOption elements within this listbox */
+  @ContentChildren(CdkOption, {descendants: true}) _options: QueryList<CdkOption>;
+
+  /** On click handler of this listbox, updates selected value of clicked option */
+  onClickUpdateSelectedOption($event: MouseEvent) {
+    this._options.toArray().forEach(option => {
+      const optionId = option.getOptionId();
+      if ($event.target instanceof Element &&
+          optionId === $event.target?.getAttribute('data-optionid')) {
+        this._updateSelectedOption(option);
+      }
+    });
+  }
+
+  private _updateSelectedOption(option: CdkOption): void {
```

Comment:
nit: I don't think you need to specify the return type if it can be easily inferred 
(here and elsewhere)
```suggestion
  private _updateSelectedOption(option: CdkOption) {
```